fvm/evm/emulator/emulator.go

```diff
@@ -384,6 +384,7 @@ func (proc *procedure) deployAt(
 
 	proc.state.SetCode(addr, ret)
 	res.DeployedContractAddress = to
+	res.ReturnedValue = to[:] // assign address to return value since it's the only returned value
```

Comment:
I have one concern about making this change here. Right now, a contract deployment results in the following `evm.TransactionExecuted` event payload:

```cadence
evm.TransactionExecuted(
    blockHeight: 3,
    blockHash: "0xc9a60a5503928c637fbed960b8224e8786f147d1e9a4b6245527f50df0e7fbad",
    transactionHash: "0xe414f90fea2aebd75e8c0b3b6a4a0c9928e86c16ea724343d884f40bfe2c4c6b",
    transaction: "f9015880808301e8488080b901086060604052341561000f57600080fd5b60eb8061001d6000396000f300606060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063c6888fa1146044575b600080fd5b3415604e57600080fd5b606260048080359060200190919050506078565b6040518082815260200191505060405180910390f35b60007f24abdb5865df5079dcc5ac590ff6f01d5c16edbc5fab4e195d9febd1114503da600783026040518082815260200191505060405180910390a16007820290509190505600a165627a7a7230582040383f19d9f65246752244189b02f56e8d0980ed44e7a56c0b200458caad20bb002982052fa09c05a7389284dc02b356ec7dee8a023c5efd3a9d844fa3c481882684b0640866a057e96d0a71a857ed509bb2b7333e78b2408574b8cc7f51238f25c58812662653",
    failed: false,
    vmError: "",
    transactionType: 0,
    gasConsumed: 103856,
    deployedContractAddress: "0x99466ED2E37B892A2Ee3E9CD55a98b68f5735db2",
    returnedValue: "606060405260043610603f576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063c6888fa1146044575b600080fd5b3415604e57600080fd5b606260048080359060200190919050506078565b6040518082815260200191505060405180910390f35b60007f24abdb5865df5079dcc5ac590ff6f01d5c16edbc5fab4e195d9febd1114503da600783026040518082815260200191505060405180910390a16007820290509190505600a165627a7a7230582040383f19d9f65246752244189b02f56e8d0980ed44e7a56c0b200458caad20bb0029",
    logs: ""
)
```

Where `returnedValue` holds the executed byte-code, basically what should be returned by `eth_getCode` for the deployed contract address. 

With `res.ReturnedValue = to[:]`, we will basically returned the deployed contract address for both `deployedContractAddress` & `returnedValue`. 

This results in less data being available in the emitted event. For example, the Gateway can use the current event payload and index the executed byte-code for each address.

Reply:
That is true and I would say the only way to resolve this would to add the deployed address value to the `EVM.Result` type, the other way is to revert to overwrite the `ReturnedValue` in the contract handler no in emulator. cc @ramtinms

Reply:
@m-Peter I added deployed contract address as the new field on the result type

Reply:
Left a question in the FLIP: https://github.com/onflow/flips/pull/225#discussion_r1561285117